backend/wmg/data/wmg_cube.py

```diff
@@ -36,3 +48,216 @@ def create_cell_count_cube(tdb_group: str):
         )
 
         tiledb.from_pandas(uri, df)
+
+
+def create_cubes(tdb_group):
+    create_expression_summary_cube(tdb_group)
+    create_cell_count_cube(tdb_group)
+
+
+def create_expression_summary_cube(tdb_group):
+    """
+    Create queryable cube and write to disk
+    """
+    uri = f"{tdb_group}/cube"
+    start_time = time.time()
+
+    with tiledb.scope_ctx(create_ctx()):
+        # Create cube
+        create_empty_cube(uri)
+
+        # load data
+        dims, vals = load_data_into_cube(tdb_group, uri)
+
+        logger.debug("Saving cube to tiledb")
+        with tiledb.open(uri, "w") as cube:
+            cube[tuple(dims)] = vals
+
+        logger.debug("Cube created, start consolidation")
+        tiledb.consolidate(uri)
+
+        logger.debug("Cube consolidated, start vacuumming")
+        tiledb.vacuum(uri)
+
+    logger.debug("Cube creation complete")
+    create_cube_sec = time.time() - start_time
+    logger.info(f"Big cube: time to create {create_cube_sec}, uri={uri}")
+
+
+def create_empty_cube(uri: str):
+    """ "
+    Create an empty cube with expected schema (dimensions and attributes) at given uri
+    """
+    if tiledb.array_exists(uri):
+        tiledb.remove(uri)
+
+    tiledb.Array.create(
+        uri,
+        expression_summary_schema,
+    )
+
+
+def load_data_into_cube(tdb_group, uri: str):
+    """
+    Load data from the concatenated corpus into the queryable cube
+    """
+    ctx = create_ctx()
+    start_time = time.time()
+    logger.debug(f"Start loading big cube at : {uri}")
+
+    with tiledb.open(f"{tdb_group}/var", ctx=ctx) as var:
+        gene_ontology_term_ids = var.query(dims=["gene_ontology_term_id"], attrs=[], use_arrow=False).df[:]
+    n_genes = len(gene_ontology_term_ids)
+
+    ##
+    # Reduce X
+    ##
+    big_cube_atts = cube_indexed_dims_no_gene_ontology + cube_non_indexed_dims
+    cell_labels, cube_index = make_cube_index(tdb_group, big_cube_atts)
+    n_groups = len(cube_index)
+
+    cube_sum = np.zeros((n_groups, n_genes), dtype=np.float32)
+    cube_nnz = np.zeros((n_groups, n_genes), dtype=np.uint64)
+    cube_min = np.zeros((n_groups, n_genes), dtype=np.float32)
+    cube_max = np.zeros((n_groups, n_genes), dtype=np.float32)
```

Comment:
don't need min/min fwiw

Reply:
these are actually used by coo_cube_pass1_into so I need to keep them